Update Grandstream provisioning

dmonty2 opened the pull request:
Updated provisioning of Grandstream phones:
- Address Book sync.
- Provisioning to match newer phone features.
- Memory Key provisioning updated.
- Printable Memory labels.txt and OpenOffice template.

Added Vtech phone.

Added:
- Canadian holidays.
- URL link to phone IP interface.

Partial work on address book to ldap export.  I have an export to ldap sync working but need to create an automated install/configure slapd script.

markjcrane commented:
> URL link to phone IP interface.

Problem with this is that some phones will require https and some http.

> app/devices/app_config.php
In app_languages we use label- and in the vendor functions its added automatically by the code to find the translation.

Incorrect
$vendors[$y]['functions'][$z]['label'] = "label-conference";
Correct
$vendors[$y]['functions'][$z]['label'] = "conference";

dmonty2 commented:
You don't have to merge URL link.  We needed a convenient way to access status pages on the phone itself.  A solution could be something like 192.168.0.3([http](http://192.168.0.3/)|[https](https://192.168.0.3)).  Another potential issue I can see with the link is some companies have their phones behind a NAT. 

---

...['label'] = "label-conference" is the same as sangoma, snom and yealink phones in the same file.

Thanks again for your great project!

dmonty2 commented:
Typo on my pull app/devices/app_config.php:1866
$vendors[$y]['functions'][$z]['value'] = "conference";

dmonty2 commented:
Also noticed that there was a newline at the end my app/devices/app_config.php that would break php.

dmonty2 commented:
I'm going to read over the changes and re-create another pull request.

dmonty2 commented on app/devices/app_config.php:
```diff
@@ -2243,4 +2285,4 @@
 		$vendors[$y]['functions'][$z]['groups'][] = "superadmin";
 		$vendors[$y]['functions'][$z]['groups'][] = "admin";
 
-?>
```
Do not merge this last newline change.  Can't figure out why git keeps adding it.

dmonty2 commented on app/provision/app_config.php:
```diff
@@ -442,5 +442,269 @@
 
-?>
\ No newline at end of file
+?>
```
Do not merge this newline.  Again can't figure out why git keeps adding it.

dmonty2 commented:
I've updated device_edit.php to include http and https links.  Fixed one typo.  Added gxp16xx provision profile.

There are two files that git has added newlines at the end.  I can't figure out how to fix this.  Please don't merge the last line of these two files:
app/devices/app_config.php
app/provision/app_config.php

markjcrane commented:
The dual link http, https is a good idea. If you are still around can you call FusionPBX support number to discuss this a little more 208-906-8227.
